jupyter/d2l-java/chapter_multilayer-perceptrons/mlp-djl.ipynb

```diff
@@ -0,0 +1,323 @@
+    "The second is our output layer."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "attributes": {
+     "classes": [],
+     "id": "",
+     "n": "5"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "SequentialBlock net = new SequentialBlock();\n",
+    "net.add(Blocks.batchFlattenBlock(784));\n",
+    "net.add(Linear.builder().setOutChannels(256).build());\n",
+    "net.add(Activation::relu);\n",
+    "net.add(Linear.builder().setOutChannels(10).build());\n",
+    "net.setInitializer(new NormalInitializer());"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that DJL, as usual, automatically\n",
+    "infers the missing input dimensions to each layer.\n",
+    "\n",
+    "The training loop is *exactly* the same\n",
+    "as when we implemented softmax regression.\n",
+    "This modularity enables us to separate \n",
+    "matters concerning the model architecture\n",
+    "from orthogonal considerations."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "int batchSize = 256;\n",
+    "int numEpochs = 10;\n",
+    "double[] trainLoss;\n",
+    "double[] testAccuracy;\n",
+    "double[] epochCount;\n",
+    "double[] trainAccuracy;\n",
+    "\n",
+    "trainLoss = new double[numEpochs];\n",
+    "trainAccuracy = new double[numEpochs];\n",
+    "testAccuracy = new double[numEpochs];\n",
+    "epochCount = new double[numEpochs];\n",
+    "\n",
+    "FashionMnist trainIter = FashionMnist.builder()\n",
+    "                            .optUsage(Dataset.Usage.TRAIN)\n",
+    "                            .setSampling(batchSize, true)\n",
+    "                            .build();\n",
+    "\n",
+    "\n",
+    "FashionMnist testIter = FashionMnist.builder()\n",
+    "                            .optUsage(Dataset.Usage.TEST)\n",
+    "                            .setSampling(batchSize, true)\n",
+    "                            .build();\n",
+    "\n",
+    "trainIter.prepare();\n",
+    "testIter.prepare();\n",
+    "\n",
+    "for(int i = 0; i < epochCount.length; i++) {\n",
+    "    epochCount[i] = (i + 1);\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "// function to collect results from metrics.\n",
+    "public void collectResults(Trainer trainer, Evaluator evaluator) {\n",
+    "    Metrics metrics = trainer.getMetrics();\n",
+    "    boolean hasTrainMetric = metrics.hasMetric(EvaluatorTrainingListener\n",
+    "                .metricName(evaluator, EvaluatorTrainingListener.TRAIN_EPOCH));\n",
+    "\n",
+    "    boolean hasValidateMetric = metrics.hasMetric(EvaluatorTrainingListener\n",
+    "                .metricName(evaluator, EvaluatorTrainingListener.VALIDATE_EPOCH));\n",
+    "\n",
+    "    switch (evaluator.getName()) {\n",
```

Comment:
You can also clean up this switch. There are two options I see: you can handle it statically but would need a Block.getEvaluator(evaluatorName). Then you can simply initialize the arrays with the whole function.

The other option is to do it dynamically and create a `Map<String, float[]> evaluatorMetrics` with the metric names as the key. Then, you can access it by metric name for building the table of data.